domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/AmSwitchValidator.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.hmcts.ccd.definition.store.domain;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.InvalidPropertyException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AmSwitchValidator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AmSwitchValidator.class);
+
+    public void validateAmPersistenceSwitchesIn(ApplicationParams appParams) {
+        List<String> duplicateForWriteDestinations = new ArrayList<>();
+        List<String> allForWriteDestinations = new ArrayList<>();
+
+        findAndStoreDuplicates(appParams.getCaseTypesWithAmWrittenOnlyToCcd(), allForWriteDestinations,
+                duplicateForWriteDestinations);
+
+        findAndStoreDuplicates(appParams.getCaseTypesWithAmWrittenOnlyToAm(), allForWriteDestinations,
+                duplicateForWriteDestinations);
+
+        findAndStoreDuplicates(appParams.getCaseTypesWithAmWrittenToBoth(), allForWriteDestinations,
+                duplicateForWriteDestinations);
+
+        List<String> duplicateForReadSources = new ArrayList<>();
+        List<String> allForReadSources = new ArrayList<>();
+
+        findAndStoreDuplicates(appParams.getCaseTypesWithAmReadFromCcd(), allForReadSources, duplicateForReadSources);
+
+        findAndStoreDuplicates(appParams.getCaseTypesWithAmReadFromAm(), allForReadSources, duplicateForReadSources);
+
+        if (!(duplicateForReadSources.isEmpty() && duplicateForWriteDestinations.isEmpty())) {
+            LOGGER.error("Duplicate configuration(s) detected for case type(s)!");
+            LOGGER.error("{} Case Types With Duplicate Read Source Configurations: {}", duplicateForReadSources.size(),
+                    duplicateForReadSources);
+            LOGGER.error("{} Case Types With Duplicate Write Destination Configurations: {}",
+                    duplicateForWriteDestinations.size(), duplicateForWriteDestinations);
+            throw new InvalidPropertyException(ApplicationParams.class, "",
```

Comment:
We could pass the property name from `application.properties` as a second argument that would indicate which of the configs is actually failing for duplicate values.

Reply:
See https://github.com/hmcts/ccd-data-store-api/blob/RDM-4866/src/main/java/uk/gov/hmcts/ccd/data/caseaccess/AMSwitch.java